### PR TITLE
nimble/gatts: Add check for RFU bits

### PR DESCRIPTION
**Pull request by Roshan23699**
Add Reserved For Future use bits check(needed for PTS)

### CODE REVIEW

**Comment by Roshan23699** [comment]
@sjanc PTAL

**Review by sjanc — CHANGES_REQUESTED** [review]
@KKopyscinski could you have a look at ble_gatts_peer_cl_sup_feat_update usage?

there is something fishy with that code,   firstly it probably shouldn't be in public gatt header, secondly,  returned value is used at ATT code while it return BLE_HS codes

This should be cleaned up, maybe simply returning BLE_ATT_ERR_UNLIKELY and/or BLE_ATT_ERR_INSUFFICIENT_RES instead of BLE_HS would be enough?

File: nimble/host/src/ble_gatts.c [hunk 1]
```diff
@@ -1587,6 +1587,8 @@ ble_gatts_peer_cl_sup_feat_update(uint16_t conn_handle, struct os_mbuf *om)
     int rc = 0;
     int feat_idx = 0;
     int i;
+    /* decimal(7) = binary(0000 0111) */
+    uint8_t rfu_mask = 7;
```
[the remark this answers is not in the document]
Comment (Roshan23699) [reply nimble/host/src/ble_gatts.c]: This, macro I defined just below BLE_GATT_CHR_CLI_SUP_FEAT_SZ in ble_gatt_priv.h file. 
I cannot define this in gatt service file(ble_svc_gatt.h) as the services are not accessible in src/host.

File: nimble/host/src/ble_gatts.c [hunk 2]
```diff
@@ -1618,10 +1620,15 @@ ble_gatts_peer_cl_sup_feat_update(uint16_t conn_handle, struct os_mbuf *om)
              */
             if (conn->bhc_gatt_svr.peer_cl_sup_feat[feat_idx] >
                 om->om_data[i]) {
-                rc = BLE_HS_EINVAL;
+                rc = BLE_ATT_ERR_VALUE_NOT_ALLOWED;
```
[the remark this answers is not in the document]
Comment (Roshan23699) [reply nimble/host/src/ble_gatts.c:1621]: Returning all att errors should work. Could you take a look again? I converted all the HS errors to ATT.

**Review by sjanc — APPROVED** [review]

**Pull request merged**